tests/integration/pilot/vm_test.go

```diff
@@ -65,3 +76,113 @@ func TestVmOSPost(t *testing.T) {
 			}
 		})
 }
+
+func TestVMRegistrationLifecycle(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresSingleCluster().
+		Features("vm.autoregistration").
+		Run(func(ctx framework.TestContext) {
+			scaleDeploymentOrFail(ctx, "istiod", i.Settings().SystemNamespace, 2)
```

Comment:
I am a bit concerned by this. I think it will work in our CI but fail in many environments, where scaling cannot be controlled like this (HPA is the typical case, or external istiod). For the latter case we can probably just skip it though